aospy/data_loader.py

```diff
@@ -0,0 +1,416 @@
+            internal_names.TIME_STR
+
+        Returns
+        -------
+        Dataset
+        """
+        ds = times.numpy_datetime_workaround_encode_cf(ds)
+        if internal_names.TIME_BOUNDS_STR in ds:
+            ds = times.set_average_dt_metadata(ds)
+        ds = xr.decode_cf(
+            ds, decode_times=True, decode_coords=False, mask_and_scale=False
+        )
+        return ds
+
+    @staticmethod
+    def _sel_var(ds, var):
+        """Search a Dataset for the specified variable, trying all possible
+        alternative names.
+
+        Parameters
+        ----------
+        ds : Dataset
+            Dataset possibly containing var
+        var : aospy.Var
+            Variable to find data for
+
+        Returns
+        -------
+        DataArray
+
+        Raises
+        ------
+        KeyError
+             If the variable is not in the Dataset
+        """
+        for name in var.names:
+            try:
+                da = ds[name]
+                return da.rename({name: var.name})
+            except KeyError:
+                pass
+        raise KeyError('{0} not found in '
+                       'among names:{1} in {2}'.format(var, var.names, ds))
+
+    @staticmethod
+    def rename_grid_attrs(ds):
+        """Rename existing grid attributes to be consistent with
+        aospy conventions.
+
+        This function does not compare to Model coordinates or
+        add missing coordinates from Model objects.  Grid attributes
+        are set as coordinates, such that they are carried by all
+        selected DataArrays with overlapping index dimensions.
+
+        Parameters
+        ----------
+        ds : Dataset
+
+        Returns
+        -------
+        renamed : Dataset
+            Dataset returned with coordinates consistent with aospy
+            conventions
+        """
+        for name_int, names_ext in internal_names.GRID_ATTRS.items():
+            ds_coord_name = set(names_ext).intersection(set(ds.coords) |
+                                                        set(ds.data_vars))
+            if ds_coord_name:
+                ds.rename({ds_coord_name.pop(): name_int}, inplace=True)
+        return ds
+
+    @staticmethod
+    def set_grid_attrs_as_coords(ds):
+        """Set available grid attributes as coordinates in a given Dataset.
+
+        Grid attributes are assumed to have their internal aospy names.
+
+        Parameters
+        ----------
+        ds : Dataset
+            Input data
+
+        Returns
+        -------
+        Dataset
+            Dataset with grid attributes set as coordinates
+        """
+        int_names = internal_names.GRID_ATTRS.keys()
+        grid_attrs_in_ds = set(int_names).intersection(set(ds.coords) |
+                                                       set(ds.data_vars))
+        ds.set_coords(grid_attrs_in_ds, inplace=True)
+        return ds
+
+    def _generate_file_set(self, var=None, start_date=None, end_date=None,
```

Comment:
This related to the issue I referenced last night in Calc.  Since at the moment Calc always provides all of those arguments, we can't limit the arguments passed to ```_generate_file_set```, otherwise we'll get a ```TypeError``` due to the specification of additional, undefined keyword arguments.  So for now I think it might be best to be explicit here (since we can't shorten the call signatures of the overriding versions).